### ISSUE TITLE
Using a script name as a variable name gives a compile error when done directly in a script, but not when done anywhere else

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#4885]
### Description
Doing global.Script1 = 0 in a script while having a script called Script1 will give a compile error, saying "Script: Script1 at line 2 : cannot use function/script name for a variable, using "Script1""
However, doing that same line anywhere else will not give a compile error. Even wrapping it in a function and calling that function inside the same script will not give the error
```
function test() {
       global.Script1 = 0;
       Script1 = 0;
}
test();
``` 
doing the above in a script will compile just fine, even though it's the same thing in the end.
I don't know if the bug is the function/object versions compiling, or the script version not compiling, but either way it's inconsistent. If the bug is that doing this in locations other than a script should give a compile error, then #4886 would be resolved as well when fixed.

### Steps To Reproduce
Create a script. Put the following inside:
global.Script1 = 0;
Script1 = 0;
Try to run the game, and you will get a compile error.
Now, put the same code inside a function in the same script and call that function. The game will now compile just fine.
Putting the code in an object event will also compile.

### Which version of GameMaker are you reporting this issue for?
IDE v2024.2.0.132 Runtime v2024.2.0.163

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22621.0

[5b688277-d526-4d11-8b8f-1eac93e95fb7](https://api.gamemaker.io/api/github/downloads/5b688277-d526-4d11-8b8f-1eac93e95fb7)

**Comment by rwkay** [comment]
Fixed in 2024.4 - at least improved it will be more consistent with the Error and report it for Scripts (and sub functions within them), you will be able to use asset names within structs, but you may need to use `self.Script1` for instance to avoid the error (explicitly telling the compiler to make it a struct/instance variable)

**Comment by caitlinrmcintyre** [comment]
Verified fix in IDE v2024.400.0.525  Runtime v2024.400.0.546, thanks.
